### ISSUE TITLE
OpenAPI version 3.1.0 is not supported. when creating Copilot Declarative Agent

### ISSUE THREAD

**Issue by alexheat** [OfficeDev/microsoft-365-agents-toolkit#12476]
**Describe the bug**
When attempting to import an OpenAPI specification file, an error message is displayed stating: "OpenAPI version 3.1.0 is not supported. Use version 3.0.x."

**To Reproduce**
1. Create new App in the Teams Toolkit
2. Choose **Declarative Agent** and then **Add Plugin** and **Start with an OpenAPI Description Document**
4. Browse for a file with OpenAI schema version 3.1.0 (see attached file as a sample)
5. Upon import I get an error "OpenAPI version 3.1.0 is not supported. Use version 3.0.x."
![image](https://github.com/user-attachments/assets/16c125f3-8ea0-4a58-bbde-d91acc2dc065)

**Expected behavior**
Open AI GPTs only support Open API 3.1.0 so in order to import an existing OpenAPI description the Teams Toolkit should support the 3.1.0 schema

**Screenshots**

From Open AI GPTs: They don't support OpenAPI 3.0
<img width="250" alt="image" src="https://github.com/user-attachments/assets/5cac957e-94d4-48f1-82cd-12df899c8255">

**VS Code Extension Information (please complete the following information):**
 - OS: Mac
 - Version: v5.9.2024091807

**CLI Information (please complete the following information):**
 - OS: 14.6.1
 - Version: 3.0.2

**Additional context**
Sample file for testing attached here: [sample.json](https://github.com/user-attachments/files/17174950/sample.json)

```
{
  "openapi": "3.1.0",
  "info": {
    "title": "Get weather data",
    "description": "Retrieves current weather data for a location.",
    "version": "v1.0.0"
  },
  "servers": [
    {
      "url": "https://weather.example.com"
    }
  ],
  "paths": {
    "/location": {
      "get": {
        "description": "Get temperature for a specific location",
        "operationId": "GetCurrentWeather",
        "parameters": [
          {
            "name": "location",
            "in": "query",
            "description": "The city and state to retrieve the weather for",
            "required": true,
            "schema": {
              "type": "string"
            }
          }
        ],
        "deprecated": false
      }
    }
  },
  "components": {
    "schemas": {}
  }
}
```

**Comment by SLdragon** [comment]
Hi, @alexheat , this is a known limitation for Copilot Declarative Agent, because the backend using [OpenAPI.NET](https://github.com/microsoft/OpenAPI.NET) to parse OpenAPI spec file, which currently only support version < 3.1.0.  We will support it once it is ready

Related issue can be found here:
https://github.com/microsoft/OpenAPI.NET/issues/795

For your scenario, if you're not using OpenAPI 3.1.0 features, you can change it to 3.0 for the Copilot Declarative Agent and keep it at 3.1.0 for OpenAI GPTs.

**Comment by abms26** [comment]
Is there a document for supported OpenAPI specs by teams toolkit?

I am facing errors like this when I try to create a plugin for a declarative agent, but the error logs are not very informative:

(×) Error: No supported API is found in the OpenAPI description document. 
Reasons for API incompatibility are listed below:
GET /HttpTriggerWithOpenAPICSharp1: authorization type is not supported
GET /PromptFlowAPIHTTPTrigger: authorization type is not supported

[SpecParser.update-manifest-failed]: Unsupported schema in post /score: {"additionalProperties":{},"type":"object"}

**Comment by SLdragon** [comment]
Hi @abms26,

Please try using TTKs version 13 or later for improved compatibility with various OpenAPI specifications and enhanced project scaffolding.

Regarding authentication, Declarative Agent supports OAuth 2.0 (auth code flow), API key authentication (header/query), and Microsoft Entra ID SSO. Other authentication types are not currently supported.

https://learn.microsoft.com/en-us/microsoft-365-copilot/extensibility/api-plugin-authentication

**Comment by garrytrinder** [comment]
Just hit this issue but on the `teamsApp/validateAppPackage` action.

I have an OpenAPI spec that is v3.0.4, in VSCode this shows as an error.

![Image](https://github.com/user-attachments/assets/2d995a53-889b-4ee4-8008-9c5667a70779)

When I change it to 3.1.0 the error goes away but I then get the validation error.

![Image](https://github.com/user-attachments/assets/ac27a37b-7f45-4e33-adea-de3816cf709e)

Changing back to 3.0.4 and provisioning the agent passes the validation step.

**Comment by SLdragon** [comment]
M365 Agent Toolkit 6.x now supported OpenAPI spec 3.1.x for Declarative Agent project now, if you still have problems, feel free to create a new issue
